task/queue/queue.go

```diff
@@ -0,0 +1,80 @@
+package queue
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+
+	"github.com/panjf2000/ants/v2"
+)
+
+// TaskQueue representation of task queue.
+type TaskQueue struct {
+	goPool             *ants.Pool
+	limit              int
+	numParallelRequest int
+
+	mutex sync.Mutex
+	queue *list.List
+}
+
+// New creates task queue which is processed by number of workers. Number of task is limited by limit.
+func New(numWorkers, limit int) (*TaskQueue, error) {
```

Comment:
Why do you want to have this in kit? Won't it be used only in the CoAP Gateway?

Reply:
I want to use it in sdk